sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3869,11 +3882,8 @@ piEnqueueMemBufferMap(pi_queue Queue, pi_mem Buffer, pi_bool BlockingMap,
   // TODO: we don't implement read-only or write-only, always read-write.
   // assert((map_flags & CL_MAP_READ) != 0);
   // assert((map_flags & CL_MAP_WRITE) != 0);
-  assert(Buffer);
   assert(Queue);
-
-  // Lock automatically releases when this goes out of scope.
-  std::lock_guard<std::mutex> lock(Queue->PiQueueMutex);
```

Comment:
why do you remove this important MT-safe logic?

Reply:
Mistake.